hedera-mirror-rosetta/app/services/construction/construction_service_test.go

```diff
@@ -0,0 +1,457 @@
+	return &types.Operation{
+		OperationIdentifier: &types.OperationIdentifier{
+			Index: index,
+		},
+		Type: transferType,
+		Account: &types.AccountIdentifier{
+			Address: account,
+		},
+		Amount: &types.Amount{
+			Value: amount,
+			Currency: &types.Currency{
+				Symbol:   "HBAR",
+				Decimals: 8,
+				Metadata: map[string]interface{}{
+					"issuer": "Hedera",
+				},
+			},
+		},
+	}
+}
+
+func networkIdentifier() *types.NetworkIdentifier {
+	return &types.NetworkIdentifier{
+		Blockchain: "SomeBlockchain",
+		Network:    "SomeNetwork",
+		SubNetworkIdentifier: &types.SubNetworkIdentifier{
+			Network: "SomeSubNetwork",
+		},
+	}
+}
+
+func dummyConstructionHashRequest(txHash string) *types.ConstructionHashRequest {
+	return &types.ConstructionHashRequest{
+		NetworkIdentifier: networkIdentifier(),
+		SignedTransaction: txHash,
+	}
+}
+
+func dummyConstructionParseRequest(txHash string, signed bool) *types.ConstructionParseRequest {
+	return &types.ConstructionParseRequest{
+		NetworkIdentifier: networkIdentifier(),
+		Signed:            signed,
+		Transaction:       txHash,
+	}
+}
+
+func dummyPayloadsRequest(operations []*types.Operation) *types.ConstructionPayloadsRequest {
+	return &types.ConstructionPayloadsRequest{
+		Operations: operations,
+	}
+}
+
+func TestNewAccountAPIService(t *testing.T) {
+	assert.IsType(t, &ConstructionAPIService{}, NewConstructionAPIService())
+}
+
+func TestNewConstructionAPIService(t *testing.T) {
+	constructionService := NewConstructionAPIService()
+	assert.IsType(t, &ConstructionAPIService{}, constructionService)
+}
+
+func TestConstructionCombine(t *testing.T) {
+	// given:
+	expectedConstructionCombineResponse := &types.ConstructionCombineResponse{
+		SignedTransaction: validTxHash,
+	}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, dummyConstructionCombineRequest())
+
+	// then:
+	assert.Equal(t, expectedConstructionCombineResponse, res)
+	assert.Nil(t, e)
+}
+
+func TestConstructionCombineThrowsWhenNoSingleSignature(t *testing.T) {
+	// given:
+	exampleConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleConstructionCombineRequest.Signatures = []*types.Signature{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.MultipleSignaturesPresent], e)
+}
+
+func TestConstructionCombineThrowsWhenDecodeStringFails(t *testing.T) {
+	// given:
+	exampleInvalidTxHashConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidTxHashConstructionCombineRequest.UnsignedTransaction = invalidTxHash
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidTxHashConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.TransactionDecodeFailed], e)
+}
+
+func TestConstructionCombineThrowsWithInvalidPublicKey(t *testing.T) {
```

Comment:
I think that you should add new test before this one with the name:
`TestConstructionCombineThrowsWhenUnmarshallFails`